### DISCUSSION TITLE
Solar midnight.

### DISCUSSION THREAD

**Discussion by marius-sucan** [jpb10/SolarCalculator#8] [General]
Hello!

I analyzed the code and found that this line:

m[0] = jd.m + wrapTo180(ra - longitude - GMST) / 360;

Is used to calculate the transit [solar noon]. I tried adjusting it to return solar midnight , but without succes. I would add m[4] for solar midnight, but my astronomy know how is too limited for this.

Could you please tell me how to get it to return the solar midnight?

Thank you .

Best regards, Marius.

**Comment by jpb10** [comment]
An approximation is transit +12 hours.

Otherwise, I think you can get the solar midnight by calculating the solar noon for your local longitude -180 degrees:

`calcSunriseSunset(utc, latitude, longitude-180, midnight, dummy, dummy);`

**Comment by jpb10** [comment]
I will convert this issue to a discussion.

**Comment by marius-sucan** [comment]
Thank you .  It seems to yield the correct results. I have tested it in polar regions as well.

Best regards, Marius.

**Comment by Macintosh-Fan** [comment]
You can probably calculate solar midnight by checking when the sun is in the opposite heading from solar noon.
